### PR TITLE
[MSAL]Add AAD error handling, refine SP logout

### PR DESCRIPTION
**Pull request by qianwens**
**Description**  

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S171

File: src/azure-cli-core/azure/cli/core/_identity.py [hunk 1]
```diff
@@ -22,6 +22,8 @@
     ManagedIdentityCredential
 )
 
+from azure.core.exceptions import ClientAuthenticationError
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_identity.py:25]: Please delay this import to void performance downgrade. See https://github.com/Azure/azure-cli/pull/13843.

File: src/azure-cli-core/azure/cli/core/_identity.py [hunk 2]
```diff
@@ -193,10 +199,13 @@ def login_with_managed_identity(self, resource, identity_id=None):
...
                         # Try object ID
-                        # TODO: Support resource ID in Azure Identity
-                        credential = ManagedIdentityCredential(object_id=identity_id)
+                        credential = ManagedIdentityCredential(identity_config={"object_id": identity_id})
+                        credential.get_token(scope)
                         id_type = self.MANAGED_IDENTITY_OBJECT_ID
                         authenticated = True
+                    except ClientAuthenticationError as e:
+                        logger.debug('MSI authentication error: %s', e.message)
+                        logger.info('Sniff: not an MSI object id')
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/_identity.py:208]: Consider using more instructive word other than `Sniff`. Also `MSI` is a deprecated name. Let's use `Managed Identity`.

**Pull request merged**